xonsh/completer.py

```diff
@@ -425,38 +427,40 @@ def bash_complete(self, prefix, line, begidx, endidx):
         else:
             prefix = shlex.quote(prefix)
 
-        script = BASH_COMPLETE_SCRIPT.format(filename=fnme,
-                    line=' '.join(shlex.quote(p) for p in splt),
-                    comp_line=shlex.quote(line), n=n, func=func, cmd=cmd,
-                    end=endidx + 1, prefix=prefix, prev=shlex.quote(prev))
+        script = BASH_COMPLETE_SCRIPT.format(
+            filename=fnme, line=' '.join(shlex.quote(p) for p in splt),
+            comp_line=shlex.quote(line), n=n, func=func, cmd=cmd,
+            end=endidx + 1, prefix=prefix, prev=shlex.quote(prev))
         try:
-            out = subprocess.check_output(['bash'], input=script,
-                    universal_newlines=True, stderr=subprocess.PIPE,
-                    env=builtins.__xonsh_env__.detype())
+            out = subprocess.check_output(
+                ['bash'], input=script, universal_newlines=True,
+                stderr=subprocess.PIPE, env=builtins.__xonsh_env__.detype())
         except subprocess.CalledProcessError:
             out = ''
 
         rtn = set(out.splitlines())
         return rtn
 
-    def _source_completions(self):
-        srcs = []
-        for f in builtins.__xonsh_env__.get('BASH_COMPLETIONS'):
-            if os.path.isfile(f):
-                # We need to "Unixify" Windows paths for Bash to understand
-                if ON_WINDOWS:
-                    f = RE_WIN_DRIVE.sub(lambda m: '/{0}/'.format(m.group(1).lower()), f).replace('\\', '/')
-                srcs.append('source ' + f)
-        return srcs
+    @staticmethod
+    def _collect_completions_sources():
+        sources = []
+        paths = (Path(x) for x in
+                 builtins.__xonsh_env__.get('BASH_COMPLETIONS'))
+        for path in paths:
+            if path.is_file():
+                sources.append('source ' + str(path))
+            elif path.is_dir():
+                for _file in (x for x in path.glob('*') if x.is_file()):
+                    sources.append('source ' + str(_file))
+        return sources
 
     def _load_bash_complete_funcs(self):
         self.bash_complete_funcs = bcf = {}
-        inp = self._source_completions()
-        if len(inp) == 0:
+        inp = self._collect_completions_sources()
+        if not inp:
```

Comment:
thanks for the research. never thought about that aspect. what results shows `not len(x)`? ( i currently bugged my xonsh)